src/commands/scan.ts

```diff
@@ -0,0 +1,141 @@
+import {Command, flags} from '@oclif/command';
+import {Scanner, ScanResult} from '../scanner';
+import {ResourceGraphTarget, DummyTarget, Target, RuleType} from '../rules';
+import cli from 'cli-ux';
+import chalk = require('chalk');
+
+interface LogOptions {
+  color?: string;
+  indent?: number;
+  bold?: boolean;
+}
+
+export default class Scan extends Command {
+  private isDebugMode = false;
+  private isVerbose = false;
+
+  static description =
+    'Scans Azure resources for potential configuration issues';
+
+  static examples = [
+    `$ aza scan --scope <SCOPE>
+    [rule-name]
+        [✓ | ❌][rule-description]     
+        Resources:
+                [resource-ids]
+
+    [total-passing]
+    [total-failing]
+    [total-rules-scanned]   
+`,
+  ];
+
+  static flags = {
+    help: flags.help({char: 'h'}),
+    scope: flags.string({
+      char: 's',
+      description: 'Azure subscription id to scan',
+    }),
+    dummy: flags.boolean({
+      char: 'd',
+      description: 'runs dummy rules to mock multi rule system',
+    }),
+    file: flags.string({
+      char: 'f',
+      description: 'JSON rules file path',
+    }),
+    verbose: flags.boolean({
+      char: 'v',
+      description: 'prints all results',
+    }),
+    debug: flags.boolean({
+      description: 'prints debugging logs',
+    }),
+  };
+
+  public log(message: string, options?: LogOptions) {
+    let formattedMessage = message;
+    if (options?.color) {
+      formattedMessage = chalk.keyword(options.color)(formattedMessage);
+    }
+    if (options?.indent) {
+      formattedMessage = ' '.repeat(options.indent) + formattedMessage;
+    }
+    if (options?.bold) {
+      formattedMessage = chalk.bold(formattedMessage);
+    }
+    super.log(formattedMessage);
+  }
+
+  private printResult(result: ScanResult) {
+    const redX = '\u274c';
+    const greenCheck = chalk.green('\u2713');
+    const totalResources = result.resourceIds.length;
+    const description = chalk.grey(result.description);
```

Comment:
I am making the color changes before sending it to the log method because I want two different colors on the same line. There might be a better way of doing this but I am not sure of the best way forward